oauth-common/src/main/java/io/strimzi/kafka/oauth/common/OAuthAuthenticator.java

```diff
@@ -0,0 +1,106 @@
+package io.strimzi.kafka.oauth.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static io.strimzi.kafka.oauth.common.TokenIntrospection.introspectAccessToken;
+
+public class OAuthAuthenticator {
+
+    private static final Logger log = LoggerFactory.getLogger(OAuthAuthenticator.class);
+
+    public static TokenInfo loginWithAccessToken(String token) {
+        log.warn("loginWithAccessToken() - pass-through access_token: " + token);
+
+        // try introspect token
+        try {
+            return introspectAccessToken(token);
+        } catch (Exception e) {
+            log.info("[IGNORED] Could not parse token as JWT access token. Could not extract scope, subject, and expiry.", e);
+        }
+        return new TokenInfo(token, "undefined", "undefined", System.currentTimeMillis(), System.currentTimeMillis() + 365 * 24 * 3600000);
+    }
+
+    public static TokenInfo loginWithClientSecret(URI tokenEndpointUrl, String clientId, String clientSecret) throws IOException {
+        log.warn("loginWithClientSecret() - tokenEndpointUrl: " + tokenEndpointUrl
+                + ", clientId: " + clientId  + ", clientSecret: " + clientSecret);
+
+        String authorization = "Basic " + base64encode(clientId + ':' + clientSecret);
+
+        StringBuilder body = new StringBuilder("grant_type=client_credentials");
+
+        return post(tokenEndpointUrl, authorization, body.toString());
+    }
+
+    public static TokenInfo loginWithRefreshToken(URI tokenEndpointUrl, String refreshToken, String clientId, String clientSecret) throws IOException {
+        log.warn("loginWithRefreshToken() - tokenEndpointUrl: " + tokenEndpointUrl + ", refreshToken: " + refreshToken
```

Comment:
The client secret can be used to get the tokens so I do notj think it should be printed in the logs.